Branch-SDK/src/io/branch/referral/RemoteInterface.java

```diff
@@ -220,6 +222,10 @@ private ServerResponse make_restful_get(String baseUrl, String tag, int timeout,
         } catch (IOException ex) {
             if (log) PrefHelper.Debug(getClass().getSimpleName(), "IO exception: " + ex.getMessage());
             return new ServerResponse(tag, 500);
+        } finally {
+            if (httpClient != null && httpClient.getConnectionManager() != null) {
```

Comment:
You could do most of this for the two functions, really all you need to do is construct the Request outside and pass it in.